libs/utils/src/utils/gas.ts

```diff
@@ -23,3 +48,73 @@ export const getProcessingGasLimit = (
 };
 
 // moved getGasCostEstimate function to utils/ethersFallback.ts
+export type Unit = 'ether' | 'gwei' | 'wei' | number;
+
+export type FetchFeeDataArgs = {
+  // Chain id to use for Public Client.
+  chainId: ValidNetwork;
+  // Units for formatting output
+  formatUnits?: Unit;
+};
+
+export function getUnit(unit: Unit) {
+  if (typeof unit === 'number') return unit;
+  if (unit === 'wei') return 0;
+  return Math.abs(weiUnits[unit]);
+}
+
+export type FetchFeeDataResult = {
+  lastBaseFeePerGas: bigint | null;
+  gasPrice: bigint | null;
+  maxFeePerGas: bigint | null;
+  maxPriorityFeePerGas: bigint | null;
+  formatted: {
+    gasPrice: string | null;
+    maxFeePerGas: string | null;
+    maxPriorityFeePerGas: string | null;
+  };
+};
+
+export async function fetchFeeData({
+  chainId,
+  formatUnits: units = 'gwei',
+}: FetchFeeDataArgs): Promise<FetchFeeDataResult> {
+  const client = createPublicClient({
+    chain: VIEM_CHAINS[chainId],
+    transport: http(),
+  });
```

Comment:
should createViemClient be moved out of abi utils into a lower level so it cna be used here in utils itself?

https://github.com/HausDAO/monorepo/blob/2762c279e351021a893f5f80ef3972f55bcc5308/libs/tx-builder/src/utils/abi.ts#L87

that should help with using publicClient in other places IF `@daohaus/keychain-utils` is also at a similar non conflicting lib level as utils itself

https://github.com/HausDAO/monorepo/blob/2762c279e351021a893f5f80ef3972f55bcc5308/libs/tx-builder/src/utils/abi.ts#L81